### PR TITLE
Detector-Competition-Feat: Add ipinfo.io API key detector

### PR DESCRIPTION
**Pull request by lc**
### Description:
This adds detection for ipinfo.io. The API documentation can be found [here](https://ipinfo.io/developers)

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### CODE REVIEW

File: pkg/detectors/ipinfo/ipinfo.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	client *http.Client
+}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	defaultClient = common.SaneHttpClient()
+	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
+	keyPat = regexp.MustCompile(`\b([a-f0-9]{14})\b`)
```
Comment (zricethezav) [inline pkg/detectors/ipinfo/ipinfo.go:25]: @lc this is too generic and will match a lot of false positives. `detectors.PrefixRegex([]string{"ipinfo"})` would tighten it up a bit

Comment (lc) [reply pkg/detectors/ipinfo/ipinfo.go]: Hey @zricethezav,

Done, thanks

**Comment by zricethezav** [comment]
@lc to remain fair to the competition, would you mind providing a screenshot of the test passing for this detector?

**Comment by lc** [comment]
CleanShot 2023-10-19 at 10 48 44

Fixing merge conflicts right now

**Pull request merged**